app/code/Magento/Search/Test/Unit/Ui/Component/Listing/Column/SynonymActionsTest.php

```diff
@@ -0,0 +1,122 @@
+    {
+        $this->urlBuilderMock = $this->createMock(UrlInterface::class);
+
+        $objectManager = new ObjectManager($this);
+
+        $this->synonymActions = $objectManager->getObject(
+            SynonymActions::class,
+            [
+                'urlBuilder' => $this->urlBuilderMock,
+                'data' => [
+                    'name' => 'actions'
+                ]
+            ]
+        );
+    }
+
+    /**
+     * Test prepareDataSource() with data source has no item
+     */
+    public function testPrepareDataSourceWithNoItem()
+    {
+        $dataSource = [
+            'data' => []
+        ];
+        $expected = [
+            'data' => []
+        ];
+        /**
+         * Assert Result
+         */
+        $this->assertEquals($expected, $this->synonymActions->prepareDataSource($dataSource));
+    }
+
+    /**
+     * Test prepareDataSource() with data source has items
+     */
+    public function testPrepareDataSourceWithItems()
+    {
+        $dataSource = [
+            'data' => [
+                'items' => [
+                    [
+                        'group_id' => 1
+                    ]
+                ]
+            ]
+        ];
+        $expected = [
+            'data' => [
+                'items' => [
+                    [
+                        'group_id' => 1,
+                        'actions' => [
+                            'delete' => [
+                                'href' => 'http://localhost/magento2/admin/search/synonyms/delete/group_id/1',
+                                'label' => (string)__('Delete'),
+                                'confirm' => [
+                                    'title' => (string)__('Delete'),
+                                    'message' => (string)__(
+                                        'Are you sure you want to delete synonym group with id: %1?',
+                                        1
+                                    )
+                                ],
+                                '__disableTmpl' => true
+                            ],
+                            'edit' => [
+                                'href' => 'http://localhost/magento2/admin/search/synonyms/edit/group_id/1',
+                                'label' => (string)__('View/Edit'),
+                                '__disableTmpl' => true
+                            ]
+                        ]
+                    ]
+                ]
+            ]
+        ];
+
+        $this->urlBuilderMock->expects($this->at(0))->method('getUrl')
+            ->with(SynonymActions::SYNONYM_URL_PATH_DELETE, ['group_id' => 1])
+            ->willReturn('http://localhost/magento2/admin/search/synonyms/delete/group_id/1');
+
+        $this->urlBuilderMock->expects($this->at(1))->method('getUrl')
+            ->with(SynonymActions::SYNONYM_URL_PATH_EDIT, ['group_id' => 1])
+            ->willReturn('http://localhost/magento2/admin/search/synonyms/edit/group_id/1');
```

Comment:
I encourage you to use ValueMap: https://phpunit.de/manual/6.5/en/test-doubles.html#idm140270329770848

```php
        $stub->method('doSomething')
             ->will($this->returnValueMap($map));
```